### PR TITLE
Remove flaky arm64 test job

### PR DESCRIPTION
**Pull request by suneet-s**
This removes a flaky test job that was added in #10562 

The travis job was added to test building Druid on Arm64 architecture. No tests are actually run as part of the job. 

However this job appears to fail around half of the time. My limited googling has not yielded any promising results. Since this impacts dev productivity, I propose we remove this job until we find out why this test fails so often and fix it appropriately.

### CODE REVIEW

**Comment by suneet-s** [comment]
@nishantmonu51 @martin-g @himanshug FYI since this is reverting a change that you all participated in. Any concerns with this?

**Comment by martin-g** [comment]
I will take a look at the failures at Monday!

**Comment by himanshug** [comment]
thanks, reducing transient failures is good, so it is ok to [temporarily] remove it since no issues have been filed specifically for things not working on arm64. so, +1

that said, I would let @martin-g take a crack at fixing this as there might be something systemic wrong and build failure might actually not be a false positive.

let us merge this towards the end of next week if things stay same.

**Comment by zhangyue19921010** [comment]
Nice catch. In my experience, this job 24 is a little bit flaky. This jobs is job often fails with `/home/travis/.travis/functions: line 109:  6122 Killed`. I am not sure what happens yet, but generally speaking, retry can pass.
So +1 to remove this job temporarily until we find out why this test fails so often and fix it appropriately.

**Comment by martin-g** [comment]
According to https://www.howtobuildsoftware.com/index.php/how-do/b5CN/travis-ci-home-travis-buildsh-line-41-pid-killed-exit-code-137 error 137 means `exhaustion of available system resources`. Most of the time it is memory related.

It is interesting that all the failures are in the build of the last module - `distribution`.

**Comment by zhangyue19921010** [comment]
> According to https://www.howtobuildsoftware.com/index.php/how-do/b5CN/travis-ci-home-travis-buildsh-line-41-pid-killed-exit-code-137 error 137 means `exhaustion of available system resources`. Most of the time it is memory related.
> 
> It is interesting that all the failures are in the build of the last module - `distribution`.

Maybe we can move this job into `Tests - phase 1`?  The resources of phase 1 may be more sufficient than phase 2?

**Comment by martin-g** [comment]
I haven't used stages before in TravisCI. I don't see anything in .travis.yml that configures resources for the stages.
But we can try it!
Another thing is to add `allow_failures` for ARM64 until we have more clue what is the reason for the kill.

**Comment by martin-g** [comment]
OK, I see how TravisCI stages work! IMO it would be even better to move the ARM64 job into a third/new stage so that it does not affect the other jobs.

**Comment by martin-g** [comment]
https://docs.travis-ci.com/user/common-build-problems/#my-build-script-is-killed-without-any-error - the max memory per job is 3Gb.
Is it an option to decrease `-Xmx3000m` to some smaller value (https://github.com/apache/druid/blob/9946306d4b2c16a7fc8bac97c9f4815ed4b46570/.travis.yml#L44) ?

**Comment by martin-g** [comment]
I've created https://github.com/apache/druid/pull/10958.
It uses the new AWS Gravoton2 based ARM64 nodes at TravisCI. Hopefully they will be more stable than the old ARM64 nodes.

**Comment by suneet-s** [comment]
> I've created #10958.
> It uses the new AWS Gravoton2 based ARM64 nodes at TravisCI. Hopefully they will be more stable than the old ARM64 nodes.

Thanks for the fix @martin-g! Since this job is still failing, I think it would be better to remove this job till we have a fix with some confidence that it will work. This way we can think through the fix fully instead of trying to rush the fix. I'll be sure to review your change as soon as it is ready so we can bring this test job back.

> Maybe we can move this job into Tests - phase 1? The resources of phase 1 may be more sufficient than phase 2?

@zhangyue19921010  This job used to be in phase 1, but would fail and prevent all the integration tests from running. I moved it to phase 2 so that a committer wouldn't need to manually start every job in phase 2 if the phase 1 job is flaky.

**Comment by jihoonson** [comment]
> https://docs.travis-ci.com/user/common-build-problems/#my-build-script-is-killed-without-any-error - the max memory per job is 3Gb.
> Is it an option to decrease -Xmx3000m to some smaller value

This kind of memory issue in CI requires a trials-and-errors type of experiments. You could try adjusting the max memory to fit in the container. Please check https://docs.travis-ci.com/user/reference/overview/ first and see how much memory the container has depending on the build environment setup. 

> Thanks for the fix @martin-g! Since this job is still failing, I think it would be better to remove this job till we have a fix with some confidence that it will work. This way we can think through the fix fully instead of trying to rush the fix. I'll be sure to review your change as soon as it is ready so we can bring this test job back.

**Review by jihoonson — APPROVED** [review]

**Comment by jihoonson** [comment]
Merging this PR as it blocks other PRs from getting merged.

**Pull request merged**
